optional_plugins/varianter_cit/avocado_varianter_cit/Parser.py

```diff
@@ -0,0 +1,77 @@
+from .Parameter import Parameter
+from .Parameter import Pair
+
+import re
+
+
+class Parser:
+
+    @staticmethod
+    def parse(file_object):
+        """
+        Parsing of input file with parameters and constraints
+        :param file_object: input file for parsing
+        :return: array of parameters and set of constraints
+        """
+
+        rx_parameter_name = re.compile(r"(.+?)\[")
+        rx_parameter_value = re.compile(r"\[(.+?)\]")
+        rx_constraint = re.compile(r".+?!=.+?")
+        parameters = []
+        constraints = set()
+        is_parameters = None
+
+        for line in file_object:
+            line = line.strip()
+            if not line:
+                continue
+            if line == "PARAMETERS":
+                is_parameters = True
+                continue
+            if line == "CONSTRAINTS":
+                is_parameters = False
+                continue
+            if is_parameters is None:
+                raise ValueError("Invalid file")
+            if is_parameters:
+                # Searching for parameter name
+                match = rx_parameter_name.search(line)
+                if match is None:
+                    raise ValueError('Parameter name is missing')
+                parameter_name = match.group(1).strip()
+
+                # Searching for parameter_values
+                match = rx_parameter_value.search(line)
+                if match is None:
+                    raise ValueError('Parameter values are missing')
+                parameter_values = [x.strip() for x in match.group(1).split(',')]
+                if len(parameter_values) != len(set(parameter_values)):
+                    raise ValueError('Parameter values has duplicities')
+
+                parameters.append(Parameter(parameter_name, len(parameters), parameter_values))
+            else:
+                constraints_data = line.split("||")
+                array = []
+                for constraint in constraints_data:
+                    value_index = 1
+                    if not rx_constraint.match(constraint):
+                        raise ValueError("Invalid format of constrain")
```

Comment:
IMO there's a typo here, as *constrain* is a verb, and *constraint* is a noun.

Reply:
Thanks, I missed that.